### ISSUE TITLE
Different optimizer rules for different sub-networks

### ISSUE THREAD

**Issue by mbsariyildiz** [facebookresearch/detectron2#529]
## 🚀 Feature
Separate learning rate schedule for backbone training.

## Motivation

I train Faster R-CNN with smaller backbones such as AlexNet and VGG16. I know that this repo does not support those networks. So I [forked](https://github.com/mbsariyildiz/detectron2) the repo and implemented them myself. What I observe is the following. When backbones are frozen the remaining parts of the network require high learning rates e.g. 1e-2 or more. to converge faster and generalize better. However when I also fine-tune backbones, I get higher mAP scores when the learning rate is lower e.g. 1e-3 or less. So, i think it would be nice to have different learning rates for backbones and the parameters trained from scratch. 

## Pitch

In [config/defaults.py](https://github.com/facebookresearch/detectron2/blob/master/detectron2/config/defaults.py#L486) maybe we can add another config entry:
```python
_C.SOLVER.BASE_LR_BACKBONE
```
or make learning rate as a config node:
```python
_C.SOLVER.BASE_LR = CN()
_C.SOLVER.BASE_LR.NEW_GUYS = 1e-2
_C.SOLVER.BASE_LR.BACKBONE = 1e-3
```
and [here](https://github.com/facebookresearch/detectron2/blob/master/detectron2/solver/build.py#L18) we need to add the new learning rate rule:
```python
        lr = cfg.SOLVER.BASE_LR
        if key.startswith("backbone"):
             lr = cfg.SOLVER.BASE_LR_BACKBONE
        weight_decay = cfg.SOLVER.WEIGHT_DECAY
        if key.endswith("norm.weight") or key.endswith("norm.bias"):
            weight_decay = cfg.SOLVER.WEIGHT_DECAY_NORM
        elif key.endswith(".bias"):
            # NOTE: unlike Detectron v1, we now default BIAS_LR_FACTOR to 1.0
            # and WEIGHT_DECAY_BIAS to WEIGHT_DECAY so that bias optimizer
            # hyperparameters are by default exactly the same as for regular
            # weights.
            lr = lr * cfg.SOLVER.BIAS_LR_FACTOR
            weight_decay = cfg.SOLVER.WEIGHT_DECAY_BIAS
        params += [{"params": [value], "lr": lr, "weight_decay": weight_decay}]
```

**Comment by ppwwyyxx** [comment]
You can already do it by just using this custom optimizer with the rest of detectron2. 
New backbones can be added easily without forking detectron2 as well.

With the principles in https://detectron2.readthedocs.io/notes/contributing.html#pull-requests, currently this does not look like a feature we would be interested in adding.

**Comment by mbsariyildiz** [comment]
Thanks for the feedback. I've modified the fork according to my needs. If you happen to be interested in this, I can do a PR.
Thanks.

**Comment by botcs** [comment]
Hi @mbsariyildiz 

For fair comparisons it would be an extremely helpful thing to have VGG16 backbone, do you have any progress on this?

Thanks

**Comment by mbsariyildiz** [comment]
Hello @botcs ,

Can you please have a look here https://github.com/mbsariyildiz/detectron2 ?

**Comment by botcs** [comment]
@mbsariyildiz coooool!
Thanks for sharing :)

At first glance, I have found some missing bits:
- I couldn't find reports on performance on Pascal VOC / COCO .
- Also as I understood you didn't use Imagenet pre-training for the backbones, though it may help with the LR problem you have mentioned here.

Can you correct me if I'm wrong and just skipped some details?

**Comment by mbsariyildiz** [comment]
Hey! 
- You are right, I haven't reported the performances, though I have them somewhere in my notes.
- It is possible to load ImageNet pre-trained weights. 

My fork is quite old actually. If you would like to put some effort, we can make a PR request.
